backend/IDE.BLL/Services/FileService.cs

```diff
@@ -47,6 +48,21 @@ public async Task<ICollection<FileDTO>> GetAllForProjectAsync(int projectId)
 
             return fileForProjectDtos;
         }
+        public async Task<IDictionary<string, FileDTO>> GetRangeByListOfIdAsync(ICollection<string> listOfId)
+        {
+            var filesForProject = new List<File>();
+            foreach(var id in listOfId)
+            {
+                var file = await _fileRepository.GetByIdAsync(id).ConfigureAwait(false);
```

Comment:
It's not a good idea to load files if loop. Why don't you make parallel requests?

Reply:
Thank you for pointing on this lines of code, but I have already refactored these)